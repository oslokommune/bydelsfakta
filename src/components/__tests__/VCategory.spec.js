import { mount, RouterLinkStub, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VCategory from '../VCategory';
import store from '../../store';
import setupI18n from '../../i18n';

const i18n = setupI18n();

describe('VCategory', () => {
  let wrapper = null;

  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(Vuex);
    wrapper = mount(VCategory, {
      propsData: {
        category: 'test',
        topic: 'test',
        bgImage: 'test',
        bgColor: 'black',
        txtColor: 'white',
        id: 'test',
        link: '/test',
        district: 'sagene',
      },
      localVue,
      stubs: { RouterLink: RouterLinkStub },
      i18n,
      store,
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('renders VCategory-component and finds main-container__item-class', () => {
    expect(wrapper.classes('main-container__item')).toBe(true);
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
