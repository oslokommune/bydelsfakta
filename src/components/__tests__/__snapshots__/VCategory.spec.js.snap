// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VCategory renders correctly 1`] = `
<a
  bgcolor="black"
  class="main-container__item"
  id="test"
  link="/test"
>
  <picture>
    <source
      alt="test"
      srcset="test.webp"
      type="image/webp"
    />
     
    <source
      alt="test"
      srcset="test.png"
      type="image/png"
    />
     
    <img
      alt="test"
      aria-hidden="true"
      class="oslo__img"
      src="test.png"
    />
  </picture>
   
  <span
    class="oslo__category"
    style="background-color: white;"
  >
    test
  </span>
   
  <span
    class="oslo__topic"
  >
    test
  </span>
</a>
`;
